src/Form/Type/TpayCardType.php

```diff
@@ -72,15 +73,29 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
             )
             ->add(
                 'expiration_date_year',
-                TextType::class,
+                ChoiceType::class,
                 [
                     'mapped' => false,
                     'label' => 'commerce_weavers_sylius_tpay.shop.order_summary.card.expiration_date.year',
+                    'placeholder' => 'commerce_weavers_sylius_tpay.shop.order_summary.card.expiration_date.year_placeholder',
+                    'choices' => $this->getCardValidYearsRange(),
                 ],
             )
             ->add('card', HiddenType::class)
         ;
 
         $builder->addModelTransformer($this->cardTypeDataTransformer);
     }
+
+    private function getCardValidYearsRange(): array
+    {
+        $result = [];
+        $currentYear = (int) date('Y');
+
+        foreach (range($currentYear, $currentYear + 10) as $year) {
```

Comment:
Why `+ 10`? Is it only our assumption, or the max value is documented somewhere? Nevertheless, I'd move `10` the const.